script/install-deps

```diff
@@ -29,6 +29,26 @@
 
 . "$(dirname $0)"/common.sh
 
+set -e
+
+install_packages() {
+  if installed apt-get; then
+    sudo apt-get update || die "apt-get update failed"
+  fi
+
+  # install wget if not available
+  if ! installed wget; then
```

Comment:
Looking forward, if the package list grows, it might be easier to group commands based on package manager rather than the package itself. See OpenThread's [script/bootstrap](https://github.com/openthread/openthread/blob/master/script/bootstrap), for example.

Reply:
I have refactored the code in another way for this purpose.

Reply:
Agree that we should follow the main `openthread` for code styles. But for implementation details, I think we should rather focus on code quality and potential issues. 

For this implementation of installing packages, I think OpenThread's current implementation in script/bootstrap has some drawbacks:

- OT's implementation assumes that all packages can be installed by one package management tool, whether it's `apt` or `brew`. This was true until we introduced `shfmt` at
https://github.com/openthread/openthread/blob/f8f75d6d05064fd63e81a3ad4518e5b3728303f8/script/bootstrap#L63 . `shfmt` is always installed by `brew`, which makes it confusing to have `brew install ...` code in `install_packages_apt`. 
- If `apt` is used, OT's implementation always need to execute `sudo apt-get update` before installing packages, even when no package needs to be installed. 

For the OTNS code here, the `install_package` function tries its best to install a package according to the arguments provided. There is no need of the assumption that all packages should be installed by one tool or another. 
For example, to install `shfmt`, the OTNS code can use this statement:
```bash
install_package shfmt --brew shfmt
```
It think this is more readable than putting `brew install shfmt` in `install_packages_apt`.

The `install_package` function can also be easily extended to support tools like `cargo`. 

That being said, I think it's good for us to try different implementations to see which works better. Thus, I would prefer the current implementation of this PR for installing packages. 

Thoughts? @wgtdkp @jwhui

Reply:
I agree that OpenThread's current mechanism may not be ideal. I think the most important point is that the bootstrap script works consistently on as many platforms/distributions as possible even as we update the bootstrap script. One approach involves isolating a specific set of steps for each distribution we support. I could also see an alternative approach that effectively deals with each package separately. I'm fine to experiment with different approaches until it is clear which is better overall.